consensus/enclave/impl/src/lib.rs

```diff
@@ -471,6 +472,14 @@ impl ConsensusEnclave for SgxConsensusEnclave {
             return Err(Error::InvalidLocalMembershipProof);
         }
 
+        // Sanity check - since our caller (TxManager::tx_hashes_to_block) collects all
+        // proof of memberships and root element at a time the ledger is not
+        // expected to change, we should end with the same root element for all
+        // transactions.
+        if root_element != &root_elements[0] {
+            return Err(Error::InvalidLocalMembershipRootElement);
```

Comment:
I would appreciate eyes on this particular change - is it actually correct?
CC @mfaulk

Reply:
I think so